testsuite/tests/conftest.py

```diff
@@ -875,17 +877,38 @@ def prometheus(testconfig, openshift):
     Returns an instance of Prometheus client.
     Skips the tests when Prometheus is not present in the project.
     """
-    if "prometheus" in testconfig:
-        return PrometheusClient(testconfig["prometheus"]["url"])
-
-    routes = openshift().routes.for_service('prometheus-operated')
-    if len(routes) == 0:
-        routes = openshift().routes.for_service('prometheus')
 
-    if len(routes) == 0:
-        warn_and_skip("Prometheus is not present in this project. Prometheus tests have been skipped.")
+    def _prepare_prometheus_endpoint(routes):
+        protocol = "https://" if "tls" in routes[0]["spec"] else "http://"
+        return protocol + routes[0]['spec']['host']
 
-    protocol = "https://" if "tls" in routes[0]["spec"] else "http://"
-    prometheus_url = protocol + routes[0]['spec']['host']
+    if "prometheus" in testconfig:
+        if "token" in testconfig["prometheus"]:
+            token = testconfig["prometheus"]["token"]
+        return PrometheusClient(testconfig["prometheus"]["url"], token)
+
+    openshift_monitoring = openshift(project="openshift-monitoring")
+    routes = openshift_monitoring.routes.for_service("thanos-querier")
+    if len(routes) > 0:
+        token = oc.whoami(cmd_args="-t")
+        namespace = openshift().project_name
```

Comment:
Do you think this will work even if you have no specified namespace in the settings?